File: OrchidCore/src/main/java/com/eden/orchid/api/theme/breadcrumbs/Breadcrumb.java
package com.eden.orchid.api.theme.breadcrumbs;


import com.eden.orchid.api.theme.pages.OrchidPage;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Breadcrumb {

    private OrchidPage page;
    private String title;

    public Breadcrumb(OrchidPage page) {
        this.page = page;
        this.title = page.getTitle();
    }

    public Breadcrumb(OrchidPage page, String title) {
        this.page = page;
        this.title = title;
    }

    // have both of these methods for parity with OrchidMenuItemImpl
    public boolean isActivePage(OrchidPage page) {
        return this.page == page;
    }

    public boolean isActive(OrchidPage page) {
        return isActivePage(page);
    }

    public String getLink() {
        return (page != null)
                ? page.getLink()
                : null;
    }

    public String activeClass(OrchidPage page) {
        return activeClass(page, "active");
    }

    public String activeClass(OrchidPage page, String className) {
        return (isActive(page)) ? className : "";
    }

}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/breadcrumbs/Breadcrumb.kt
package com.eden.orchid.impl.themes.breadcrumbs

import com.eden.orchid.api.theme.pages.OrchidPage

class Breadcrumb(
        val page: OrchidPage?,
        var title: String = page?.title ?: ""
) {

    val link: String?
        get() = if (page != null)
            page.link
        else
            null

    // have both of these methods for parity with OrchidMenuItemImpl
    fun isActivePage(page: OrchidPage): Boolean {
        return this.page === page
    }

    fun isActive(page: OrchidPage): Boolean {
        return isActivePage(page)
    }

    @JvmOverloads
    fun activeClass(page: OrchidPage, className: String = "active"): String {
        return if (isActive(page)) className else ""
    }

}
